dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/DynamicPureLoopCutting.java

```diff
@@ -122,22 +127,24 @@ private void insertSideEffectChecks(IterationData iter) {
 
         final List<Register> trackingRegs = new ArrayList<>();
         Event insertionPoint = iterInfo.getIterationEnd();
+        Type type = TypeFactory.getInstance().getPointerType();
         for (int i = 0; i < sideEffects.size(); i++) {
             final Event sideEffect = sideEffects.get(i);
             final Register trackingReg = thread.newRegister(
                     String.format("Loop%s_%s_%s", loopNumber, iterNumber, i),
-                    GlobalSettings.getArchPrecision());
+                    type);
             trackingRegs.add(trackingReg);
 
             final Event execCheck = EventFactory.newExecutionStatus(trackingReg, sideEffect);
             insertionPoint.insertAfter(execCheck);
             insertionPoint = execCheck;
         }
 
-        final BExpr atLeastOneSideEffect = trackingRegs.stream()
-                .map(reg -> (BExpr) new Atom(reg, COpBin.EQ, IValue.ZERO))
-                .reduce(BConst.FALSE, (x, y) -> new BExprBin(x, BOpBin.OR, y));
-        final CondJump assumeSideEffect = EventFactory.newJumpUnless(atLeastOneSideEffect, (Label) thread.getExit());
+        ExpressionFactory expressionFactory = ExpressionFactory.getInstance();
+        Expression noSideEffect = trackingRegs.stream()
+                .map(reg -> expressionFactory.makeBinary(reg, COpBin.NEQ, expressionFactory.makeZero(type)))
+                .reduce(expressionFactory.makeTrue(), (x, y) -> expressionFactory.makeBinary(x, BOpBin.AND, y));
+        final CondJump assumeSideEffect = EventFactory.newJump(noSideEffect, (Label) thread.getExit());
```

Comment:
Is there any reason why you removed the convenience function `EventFactory.newJumpUnless`?
A big use-case of a factory is to have these convenience methods.